lib/onc_certification_g10_test_kit/smart_invalid_pkce_group.rb

```diff
@@ -0,0 +1,244 @@
+          name: :standalone_client_secret,
+          title: 'Standalone Client Secret',
+          description: 'Client Secret provided during registration of Inferno as a standalone application'
+        },
+        requested_scopes: {
+          name: :standalone_requested_scopes,
+          title: 'Standalone Scope',
+          description: 'OAuth 2.0 scope provided by system to enable all required functionality',
+          type: 'textarea',
+          default: %(
+            launch/patient openid fhirUser offline_access
+            patient/Medication.read patient/AllergyIntolerance.read
+            patient/CarePlan.read patient/CareTeam.read patient/Condition.read
+            patient/Device.read patient/DiagnosticReport.read
+            patient/DocumentReference.read patient/Encounter.read
+            patient/Goal.read patient/Immunization.read patient/Location.read
+            patient/MedicationRequest.read patient/Observation.read
+            patient/Organization.read patient/Patient.read
+            patient/Practitioner.read patient/Procedure.read
+            patient/Provenance.read patient/PractitionerRole.read
+          ).gsub(/\s{2,}/, ' ').strip
+        },
+        url: {
+          title: 'Standalone FHIR Endpoint',
+          description: 'URL of the FHIR endpoint used by standalone applications'
+        },
+        code: {
+          name: :invalid_token_code
+        },
+        state: {
+          name: :invalid_token_state
+        },
+        smart_authorization_url: {
+          title: 'OAuth 2.0 Authorize Endpoint',
+          description: 'OAuth 2.0 Authorize Endpoint provided during the patient standalone launch'
+        },
+        smart_token_url: {
+          title: 'OAuth 2.0 Token Endpoint',
+          description: 'OAuth 2.0 Token Endpoint provided during the patient standalone launch'
+        },
+        pkce_code_challenge: {
+          name: :invalid_token_pkce_code_challenge
+        },
+        pkce_code_verifier: {
+          name: :invalid_token_pkce_code_verifier
+        }
+      },
+      outputs: {
+        code: { name: :invalid_token_code },
+        state: { name: :invalid_token_state },
+        expires_in: { name: :invalid_token_expires_in },
+        pkce_code_challenge: { name: :invalid_token_pkce_code_challenge },
+        pkce_code_verifier: { name: :invalid_token_pkce_code_verifier }
+      },
+      requests: {
+        redirect: { name: :invalid_token_redirect },
+        token: { name: :invalid_token_token }
+      }
+    )
+
+    # no code_verifier
+
+    test from: :smart_app_redirect_stu2,
+         id: :smart_no_code_verifier_redirect
+    test from: :smart_code_received,
+         id: :smart_no_code_verifier_code_received
+
+    test from: :invalid_pkce_request do
+      title ' OAuth token exchange fails when no code_verifier is given'
+      id :smart_no_verifier_token_request
+    end
+
+    # code_verifier=
+
+    test from: :smart_app_redirect_stu2,
```

Comment:
Can we customize the wait message provided to the user so they don't see the same redirect message each time.  I would think people would think it is in an error and it is stuck in a loop? So for each, maybe prepend a line like 'Invalid PKCE code_verifier 3/4: This launch will provide a blank code_verifier and verify the server does not issue a bearer token. \n\n <and then follow with normal launch message>'.